pkg/virt-operator/util/config.go

```diff
@@ -553,6 +554,16 @@ func (c *KubeVirtDeploymentConfig) SetTargetDeploymentConfig(kv *v1.KubeVirt) er
 	return err
 }
 
+func (c *KubeVirtDeploymentConfig) SetDefaultArchitecture(kv *v1.KubeVirt) error {
+	if kv.Spec.Configuration.ArchitectureConfiguration != nil && kv.Spec.Configuration.ArchitectureConfiguration.DefaultArchitecture != "" {
+		kv.Status.DefaultArchitecture = kv.Spec.Configuration.ArchitectureConfiguration.DefaultArchitecture
+	} else {
```

Comment:
@daghaian @rmohr I wonder if this is the correct implementation of options 3 in this comment

https://github.com/kubevirt/kubevirt/pull/8293#discussion_r1145321704

An example of corner case I am worried about is:
1. kv.Spec.Configuration.ArchitectureConfiguration is nil but kv.Status.DefaultArchitecture is set to amd64
2. virt-operator gets deployed on arm64 node in a multi-cluster environment.
3. The operator re-writes status.DefaultArchitectore to arm64

I think you need to update the status only if it is empty in the else part

Reply:
This can also be a follow-up PR in order to make progress on this PR.

Reply:
@alaypatel07  Added the corner case handling :)